d2/src/main/java/com/linkedin/d2/xds/XdsClient.java

```diff
@@ -43,16 +44,22 @@ interface ResourceWatcher
   interface NodeResourceWatcher extends ResourceWatcher
   {
     void onChanged(NodeUpdate update);
+
+    void onDelete();
```

Comment:
Since the deletion is handled via the `onRemoval` logic, I think you can remove this altogether